capstone/src/main/java/com/google/sps/servlets/FileUnzipServlet.java

```diff
@@ -0,0 +1,161 @@
+import com.google.cloud.storage.StorageOptions;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ByteArrayInputStream;
+
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.Arrays;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import com.google.api.gax.paging.Page;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+
+
+@WebServlet("/unzip")
+@MultipartConfig
+public class FileUnzipServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    
+    // String userId = request.getParamter("userId");
+    //Hard-coded UserId
+    String userId = "abcde";
+    
+    // The ID of GCP project
+    String projectId = "step-2020-team-2";
+
+    // The ID of GCS bucket
+    String bucketName = "vaderker-uploadedstoragebucket";
+
+    // String fileName = request.getParamter("files");
+    //Hard-coded apkName
+    String nameOfApk = "ApiDemos-debug.apk";
+
+    // The ID of your GCS object
+    String objectName = "apks/" + nameOfApk;
+    System.out.println(objectName);
+
+    Blob blob = getApkObjectFromCloudStorage(projectId, bucketName, objectName);
+    analyzeApkFeatures(nameOfApk, blob, userId);
+
+    System.out.println("File " + nameOfApk + " uploaded to bucket " + bucketName + " as " + objectName);
+
+    response.setContentType("text/html;charset=UTF-8");
+
+  }
+  
+  public static Blob getApkObjectFromCloudStorage(String projectId, String bucketName, String objectName) {
+    
+    // Initiate bucket details from CloudStorage for apk retrieval
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    Bucket bucket = storage.get(bucketName);
+    
+    // Prints the entire apk in Cloud Storage to the console
+    Page<Blob> blobs = bucket.list();
+    for (Blob blob1 : blobs.iterateAll()) {
+      System.out.println(blob1.getName());
+    }
+
+    // Retrieve the blob for the apk 
+    Blob blob = storage.get(BlobId.of(bucketName, objectName));
+    System.out.println(blob.getName());
+    
+    return blob;
+
+   
+  }
+
+  public static void analyzeApkFeatures(String nameOfApk, Blob blob, String userId) {
+    
+    byte[] apkBytes = blob.getContent();
+    long dexFileSize = 0, resFileSize = 0, libraryFileSize  = 0, assetsFileSize = 0, resourcesFileSize = 0, miscFileSize = 0, totalApkSize = 0;
```

Comment:
I think I would normally split these into different rows. As a follow up piece of work I would probably change this to be some sort of data object rather than a bunch of local variables.

Reply:
Please split line 99 into separate lines. This is valid but it's not very readable.